File: parent/3517 -- Smallest Palindromic Rearrangement I/mine2.py
class Solution:
    def smallestPalindrome(self, s: str) -> str:
        # Shorter. Same half-sort idea
        # Time: O(nlogn)
        # Space:  O(n)
        n = len(s)

        left = sorted(s[:n // 2])
        return ''.join(left + ([s[n // 2]] if n%2 else []) + left[::-1])

File: parent/3517 -- Smallest Palindromic Rearrangement I/mine2.go
package main

import (
	"slices"
)

func smallestPalindrome(s string) string {
	n := len(s)

	first := []byte(s[:n/2])
	slices.Sort(first)

	ans := make([]byte, 0, n)
	ans = append(ans, first...)

	if n%2 == 1 {
		ans = append(ans, s[n/2])
	}

	for i := len(first) - 1; i >= 0; i-- {
		ans = append(ans, first[i])
	}

	return string(ans)
}
